src/adcm_pytest_plugin/steps/actions.py

```diff
@@ -312,7 +314,14 @@ def _run_action_and_assert_result(
             kwargs["verbose"] = options.verbose_actions  # pylint: disable=no-member
         obj.reread()
         try:
-            task = obj.action(name=action_name).run(**kwargs)
+            action = obj.action(name=action_name)
+            if getattr(options, "actions_report_dir", None):
+                pytest.action_run_storage.append(
+                    ActionRunInfo.from_action(action=action, expected_status=expected_status)
+                )
+                actions_spec = ActionsSpec.from_action(action=action)
+                pytest.actions_spec_storage[actions_spec.uniq_id] = actions_spec
```

Comment:
Maybe move it to helper function with an early exit to not make an accent of a variable feature.
Like that:
``` python
def _add_action_to_report(options, action, expected_status): 
    if not getattr(options, "actions_report_dir", None):
        return
    pytest.action_run_storage.append(
        ActionRunInfo.from_action(action=action, expected_status=expected_status)
    )
    actions_spec = ActionsSpec.from_action(action=action)
    pytest.actions_spec_storage[actions_spec.uniq_id] = actions_spec
```
of even leave options check as is and move only report logic

Reply:
Done